tests/unit/IonTextReaderTest.js

```diff
@@ -125,6 +125,29 @@
       assert.equal(ionReader.fieldName(), "innerkey2");
     };
 
+      suite['Accept IVM like symbols and throw on IVMs except $ion_1_0'] = function() {
+          let shouldPass = ["$ion_1_0", "$ion_schema_1_0", "$ion_1", "$ion_1_a", "$ion_", "ion_1_"];
+          let shouldFail = ["$ion_2_0", "$ion_1_999", "$ion_999_0", "$ion_1_1", "$ion_1_00"];
+          for(let input in shouldPass) {
+              let ionReader = ion.makeReader(input);
+              try {
+                  while (ionReader.next() !== undefined) {}
```

Comment:
this is a unit test, so (generally speaking) we should narrow the scope of what we're invoking.  invoking `ionReader.next()` seems fairly indirect.  is there any reason not to call `isIVM()` directly?

Reply:
I think if the testing suites delineated between unit tests and integration tests I'd agree. I don't think .next() is a indirect in the context of an integration test. The intention is making the reader and parser is validate input correctly. I think a separate unit test could add value but this isn't intended to be a true unit test.

Reply:
i think a unit test makes sense here;  help me understand why you're thinking it should be an integration test?  (and if that's the approach, how does it verify that `isIVM()` returns true or false when it is expected to?)

Reply:
Ah I see, if the isIVM function throwing was your original issue I can compare the error message! easy change.